### PR TITLE
Don't always delete $NODE_DIR

### PR DESCRIPTION
**Pull request by willbarton**
This PR addresses #2654 and implements @rosskarchner's suggestion in GHE#563 to checksum `package.json`, store the checksum in `$NODE_DIR/CHECKSUM`, and then only delete $NODE_DIR if the checksums don't match. This should improve the efficiency of our `frontend.sh` script.

Some quick and dirty metrics on my local machine:

Full install of all dependencies (checksum doesn't exist or doesn't match):

```
./frontend.sh  110.17s user 30.05s system 66% cpu 3:29.80 total
```

When no dependencies have changed (checksum exists and matches):

```
./frontend.sh  55.34s user 2.62s system 155% cpu 37.200 total
```

So, this cuts down the time spent installing frontend dependencies by half unless `package.json` has changed.

## Changes

- Checksum package.json to avoid reinstalling dependencies that haven't changed

## Testing

- Run `frontend.sh` once to generate the checksum file, then run it again and observe the message:

  `Project dependencies checksum matches package.json…`
  
  And that node dependencies aren't reinstalled.

## Notes

I want to get this in front of Travis to make sure I'm not making assumptions about the availability of `shasum`. 

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

**Comment by rosskarchner** [comment]
is the npm-shrinkwrap file relevant? Are there circumstances where the shrinkwrap changes but not package.json?

**Comment by willbarton** [comment]
@rosskarchner Oh, good point. I should just use the full command you suggested 😁

**Comment by sebworks** [comment]
> Are there circumstances where the shrinkwrap changes but not package.json?

It should never happen but FEWDs do strange things.

File: frontend.sh [hunk 1]
```diff
@@ -15,25 +15,34 @@ init() {
...
+      echo 'Project dependencies have been removed.'
+    else
+      echo 'Project dependencies checksum matches package.json…'
+    fi
   fi
-  echo 'Project dependencies have been removed.'
 }
 
 # Install project dependencies.
 install() {
```
Comment (willbarton) [inline frontend.sh]: @sebworks if we're not deleting the node_modules directory and package.json and npm-shrinkwrap.json haven't changed is there a need to run `install` here at all? That could also reduce the amount of time spent.

So then we check for the existence and validity of the checksum, and if it passes we just build.

Comment (Scotchester) [inline frontend.sh]: Agreed. If `package.json` and `npm-shrinkwrap.json` haven't changed, no need to `install()`

Comment (willbarton) [inline frontend.sh]: Slicing the `install` step off if the checksums match leaves us with this:

```
./frontend.sh  37.33s user 1.92s system 120% cpu 32.684 total
```

Comment (sebworks) [inline frontend.sh]: very nice, very nice indeed.

File: frontend.sh [hunk 2]
```diff
@@ -94,14 +112,12 @@ is_installed() {
 if [ "$1" == "init" ] || [ "$1" == "build" ]; then
   if [ "$1" == "init" ]; then
     init ""
-    clean
-    install
+    clean_and_install
```
Comment (sebworks) [inline frontend.sh]: @willbarton, how would you feel about:

```

if [ "$1" == "init" ]; then
  init ""
  clean_and_install
elif [ "$1" == "build" ]; then
  build
else
  init "$1"
  clean_and_install
  build
fi
```

Comment (willbarton) [inline frontend.sh]: That works for me.

**Pull request merged**